File: src/main/scala/io/vertx/ext/asyncsql/postgresql/impl/PostgresqlConnectionImpl.scala
package io.vertx.ext.asyncsql.postgresql.impl

import com.github.mauricio.async.db.Connection
import io.vertx.core.Vertx
import io.vertx.ext.asyncsql.impl.{BaseConnectionImpl, BaseTransactionImpl}
import io.vertx.ext.asyncsql.postgresql.{PostgresqlConnection, PostgresqlTransaction}

import scala.concurrent.Future

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>.
 */
class PostgresqlConnectionImpl(val vertx: Vertx,
                               override protected val connection: Connection,
                               override protected val freeHandler: Connection => Future[_])
  extends BaseConnectionImpl with PostgresqlConnection {

}

File: vertx-mysql-postgresql-client-scala/src/main/java/io/vertx/ext/asyncsql/impl/PostgreSQLConnectionImpl.java
package io.vertx.ext.asyncsql.impl;

import com.github.mauricio.async.db.Connection;
import com.github.mauricio.async.db.QueryResult;
import io.vertx.core.json.JsonArray;
import io.vertx.ext.asyncsql.impl.pool.AsyncConnectionPool;
import io.vertx.ext.sql.UpdateResult;
import scala.concurrent.ExecutionContext;

/**
 * @author <a href="mailto:plopes@redhat.com">Paulo Lopes</a>.
 */
public class PostgreSQLConnectionImpl extends AsyncSQLConnectionImpl {
  public PostgreSQLConnectionImpl(Connection conn, AsyncConnectionPool pool, ExecutionContext ec) {
    super(conn, pool, ec);
  }

  @Override
  protected String getStartTransactionStatement() {
    // TODO: consider the tx isolation level
    return "BEGIN";
  }

    @Override
  protected UpdateResult queryResultToUpdateResult(QueryResult qr) {
    int affected = (int) qr.rowsAffected();
    return new UpdateResult(affected, new JsonArray());
  }
}
